mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -153,38 +156,33 @@ where
 
     test_client.put_object("dir/hello.txt", b"hello world").unwrap();
 
-    let path = mount_point.path().join("dir/hello.txt");
-
-    // Existing files should not be writable even in O_APPEND
-    let err = open_for_write(&path, false).expect_err("can't write existing file");
-    assert_eq!(err.kind(), ErrorKind::PermissionDenied);
-    let err = open_for_write(&path, true).expect_err("can't write existing file");
-    assert_eq!(err.kind(), ErrorKind::PermissionDenied);
+    let new_file = mount_point.path().join("dir/new.txt");
+    let existing_file = mount_point.path().join("dir/hello.txt");
 
     // New files can't be opened with O_SYNC
     let err = File::options()
         .write(true)
         .create(true)
         .custom_flags(libc::O_SYNC)
-        .open(&path)
+        .open(new_file)
         .expect_err("O_SYNC should fail");
     assert_eq!(err.kind(), ErrorKind::InvalidInput);
 
+    // Existing files can't be opened with O_APPEND
+    let err = File::options()
+        .write(true)
+        .create(true)
+        .custom_flags(libc::O_APPEND)
+        .open(&existing_file)
+        .expect_err("O_APPEND should fail");
+    assert_eq!(err.kind(), ErrorKind::PermissionDenied);
+
     // We can't write to a file opened in O_RDONLY
-    let mut file = File::options().read(true).open(&path).unwrap();
+    let mut file = File::options().read(true).open(&existing_file).unwrap();
     let err = file
         .write(b"hello world")
         .expect_err("writing to O_RDONLY file should fail");
     assert_eq!(err.raw_os_error(), Some(libc::EBADF));
-
-    // Existing files can be opened O_RDWR but only reading should work on them
-    let mut file = File::options().read(true).write(true).create(true).open(&path).unwrap();
-    let err = file
-        .write(b"hello world")
-        .expect_err("write to an existing file should fail");
-    assert_eq!(err.raw_os_error(), Some(libc::EBADF));
-    // However, read should work
-    assert!(file.read(&mut [0u8; 1]).is_ok());
```

Comment:
Are we missing a test that checks you can't read an `O_RDWR` file once it's open in write mode, and vice versa?